src/main/src/main/java/org/geoserver/catalog/ResourcePool.java

```diff
@@ -1857,26 +1857,16 @@ public void writeStyle( StyleInfo style, InputStream in ) throws IOException {
 	 *            file to update
 	 * @throws IOException
 	 */
-	public static void writeStyle(final InputStream in, final File styleFile)
+	public static void writeStyle(final InputStream in, final Resource styleFile)
 			throws IOException {
-		final File temporaryFile = File.createTempFile(styleFile.getName(),
-				null, styleFile.getParentFile());
 		BufferedOutputStream out = null;
 		try {
-			out = new BufferedOutputStream(new FileOutputStream(temporaryFile));
+			out = new BufferedOutputStream(styleFile.out());
 			IOUtils.copy(in, out);
 			out.flush();
 		} finally {
 			out.close();
 		}
-		// move the file
-		try {
-			org.geoserver.data.util.IOUtils.rename(temporaryFile, styleFile);
```

Comment:
There is no such risk because the resource store *always* does this (write to a temporary file first, then rename it when it is finished.) See: https://github.com/geoserver/geoserver/blob/master/src/platform/src/main/java/org/geoserver/platform/resource/FileSystemResourceStore.java#L226

Separation of concerns, remember ;)